azureenergylabelerlib/entities.py

```diff
@@ -724,16 +724,17 @@ def _export_to_fs(self, directory, filename, data):
     def _export_to_blob(self, blob_url, filename, data):
         """Exports as json to Blob container object storage."""
         parsed_url = urlparse(blob_url)
-        blob_service_client = BlobServiceClient(account_url=f'{parsed_url.scheme}://{parsed_url.netloc}/',
+        # This enables uploads using URLs that include a SAS token, which is the safest
+        blob_service_client = BlobServiceClient(account_url=blob_url) if parsed_url.query else BlobServiceClient(account_url=f'{parsed_url.scheme}://{parsed_url.netloc}/',
```

Comment:
Maybe use some intermediate variable here for account_url and don't try to instantiate at the same time for the sake of readability.